frappe/model/base_document.py

```diff
@@ -298,6 +355,45 @@ def _get_table_fields(self):
 
 		return self.meta.get_table_fields()
 
+	def _get_datetime_fields(self):
+		"""
+		To get datetime fields during Document init
+		"""
+		if (
+			self.doctype == "DocType"
+			or self.doctype in DOCTYPES_FOR_DOCTYPE
+			or getattr(self, "parentfield", None)
+		):
+			return ()
+
+		return self.meta.get_datetime_fields()
```

Comment:
We should do something to add the standard meta fields too. These guys (creation, modified) are missing out on the sweet casting:

https://github.com/frappe/frappe/blob/1297b25ecf08d89736fc7f04e82d07aa97f68690/frappe/model/meta.py#L254-L265

```suggestion
		return self.meta.get_datetime_fields() + self.meta.standard.get_datetime_fields() # which returns mocked DocField objects of creation & modified 
```